[panic]  runtime error: index out of range [255] with length 255

We've seen following error in our buildkite jobs: [example](https://buildkite.com/ray-project/ray-builders-branch/builds/2657#952b0a74-b8c7-4d13-86a2-b84655684a9a)

Though I'm not too familiar with go, but looks like https://github.com/buildkite/agent/blob/main/bootstrap/redactor.go#L19
 should be 256 instead of 255?

```
panic: runtime error: index out of range [255] with length 255
--
  |  
  | goroutine 81 [running]:
  | github.com/buildkite/agent/v3/bootstrap.(*Redactor).Write(0xc000400000, 0xc0004bc000, 0x21c, 0x8000, 0x21c, 0x0, 0x0)
  | /work/bootstrap/redactor.go:125 +0xdd7
  | io.copyBuffer(0xf22900, 0xc000400000, 0xf23720, 0xc00052c140, 0xc0004bc000, 0x8000, 0x8000, 0xdf4970, 0xc0000457c0, 0x762325)
  | /usr/local/go/src/io/io.go:425 +0x21c
  | io.Copy(...)
  | /usr/local/go/src/io/io.go:382
  | github.com/buildkite/agent/v3/process.(*Process).Run.func2(0xc000536e00, 0xc00052c140, 0xc000510e90, 0xc000522c40)
  | /work/process/process.go:181 +0xa9
  | created by github.com/buildkite/agent/v3/process.(*Process).Run
  | /work/process/process.go:176 +0x39c
```

--- a/bootstrap/redactor.go
+++ b/bootstrap/redactor.go
@@ -16,7 +16,7 @@ type Redactor struct {
 	maxlen int
 
 	// Table of Boyer-Moore skip distances, and values to redact matching this end byte
-	table [255]struct {
+	table [256]struct {
 		skip    int
 		needles [][]byte
 	}

--- a/bootstrap/redactor_test.go
+++ b/bootstrap/redactor_test.go
@@ -112,3 +112,11 @@ func TestRedactorSubsetSecrets(t *testing.T) {
 		t.Errorf("Redaction failed: %s", buf.String())
 	}
 }
+
+func TestRedactorLatin1(t *testing.T) {
+	var buf bytes.Buffer
+	redactor := NewRedactor(&buf, "[REDACTED]", []string{"ÿ"})
+
+	redactor.Write([]byte("foo"))
+	redactor.Flush()
+}

